### PR TITLE
added swipe gesture to book card

### PR DESCRIPTION
**Pull request by AntonKiylo**

### CODE REVIEW

File: src/redux/actions/bookActions.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const deleteBookAsync = createAsyncAction(
 
 export const addBookToBasket = createAction(
   'ADD_BOOK_TO_BASKET',
-  (id: string) => id,
+  (id?: string) => id,
```
Comment (sergej-prilutskiy) [inline src/redux/actions/bookActions.ts:30]: As far as I understand the `id` is optional here as it's coming from the component, where the `id` is also optional. That is not a good practice to alter the business logic based on the components. Adding something to the basket naturally requires that _something_, adding nothing does not make sense (and can lead to bugs). It's better to leave that clear, and simply check if the `id` is present in the component before dispatching an action

File: src/features/BookCard/BookCard.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      ownerID,
+      userID,
+    ],
+  );
+
+  return (
+    <Swipeable renderRightActions={rightSwipeAction}>
+      <View style={styles.container}>
+        <TouchableOpacity
+          onPress={() => navigation.navigate('BookScreen', {id})}>
```
[the remark this answers is not in the document]
Comment (AntonKiylo) [reply src/features/BookCard/BookCard.tsx:118]: done

File: src/redux/actions/bookActions.ts (hunk 1 above)
Comment (AntonKiylo) [reply src/redux/actions/bookActions.ts:30]: done

File: src/features/BookCard/BookCard.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  const rightSwipeAction = useCallback(
+    (_, dragX) => {
+      const scale = dragX.interpolate({
+        inputRange: [0, 30],
+        outputRange: [0, -0.6],
+      });
+
+      return (
+        <View>
+          {userID !== ownerID ? (
```
[the remark this answers is not in the document]
Comment (AntonKiylo) [reply src/features/BookCard/BookCard.tsx:77]: done

File: src/features/BookCard/BookCard.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+interface IProps {
+  id?: string;
+}
+
+const BookCard: React.FC<IProps> = ({id}) => {
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const bookID = id;
+
+  const bookTitle = useSelector((state: IState) =>
```
[the remark this answers is not in the document]
Comment (AntonKiylo) [reply src/features/BookCard/BookCard.tsx:30]: done

**Pull request merged**